File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    var numsdiff = new Map()
    var result = []
    for (let i = 0; i < nums.length; i++) {
        diff = target - nums[i]
        if (numsdiff[diff] != undefined) {
            result.push(numsdiff[diff], i)
        }
        numsdiff[nums[i]] = i
    }
    return result
};

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    numsMap := make(map[int]int)
    for i , v := range nums {  
        different := target - v
        if _ , ok := numsMap[different]; ok {
            return []int{numsMap[different] , i}
        }

        numsMap[v]  = i
    }

    return nil
}